lib/finitomata.ex

```diff
@@ -442,20 +443,21 @@ defmodule Finitomata do
           end
         else
           err ->
+            state_with_error = struct(state, error: err)
```

Comment:
`state` is already a struct, I’d prefer a more explicit
```suggestion
            state = %State{state | error: err}
```

Also, it’s still our old good state, no need to have a new name, _Elixir_ is not 100% _Erlang_.